SpatialGDK/Source/SpatialGDK/Private/Utils/InterestFactory.cpp

```diff
@@ -316,18 +327,20 @@ Interest InterestFactory::CreatePlayerOwnedActorInterest() const
 
 	if (SpatialGDKSettings->bEnableClientResultTypes)
 	{
-		ClientQuery.ResultComponentId = ClientResultType;
+		ClientQuery.ResultComponentId = ClientNonAuthInterestResultType;
 	}
 	else
 	{
 		ClientQuery.FullSnapshotResult = true;
 	}
-	
 
-	ComponentInterest ClientComponentInterest;
-	ClientComponentInterest.Queries.Add(ClientQuery);
+	AddComponentQueryPairToInterestComponent(OutInterest, ClientEndpointComponentId, ClientQuery);
 
-	AddUserDefinedQueries(LevelConstraints, ClientComponentInterest.Queries);
+	TArray<Query> UserQueries = GetUserDefinedQueries(LevelConstraints);
+	for (const auto UserQuery : UserQueries)
```

Comment:
Should be const auto& if you don't want to make copys.

Reply:
thanks, good catch